lua/flatbuffers/binaryarray.lua

```diff
@@ -0,0 +1,115 @@
+local m = {}
+
+local mt = {}
+
+function mt:__len()
+    return self.size
+end
+
+function m.New(size)
+    local o = 
+    {
+        data = {},
+        size = size
+    }
+    setmetatable(o, {__index = mt, __len = mt.__len})
+    return o
+end
+
+function m.FromString(str)
+    local o = 
+    {
+        str = str,
+        size = #str
+    }
+    setmetatable(o, {__index = mt, __len = mt.__len})
+    return o    
+end
+
+function mt:Slice(startPos, endPos)
+    startPos = startPos or 0
+    endPos = endPos or self.size
+    local d = self.data
+    if d then      
+        
+        -- new table to store the slice components
+        local b = {}        
+        
+        -- starting with the startPos, put all
+        -- values into the new table to be concat later
+        -- updated the startPos based on the size of the
+        -- value
+        while startPos < endPos do
+            local v = d[startPos] or '/0'
+            table.insert(b, v)
+            startPos = startPos + #v
+        end
+
+        -- combine the table of strings into one string
+        return table.concat(b)   
+    else
+        -- n.b start/endPos are 0-based incoming, so need to convert
+        --     correctly. in python a slice includes start -> end - 1
+        return self.str:sub(startPos+1, endPos)
```

Comment:
@aardappel When reading the buffer, I do store the buffer as a binary string and use the ;sub() function to extract parts of it.